XCode generator failure

Reported by @clevelam 

* XCode doesn't like `ds++`'s use of per file per config CPP defines.

```
CMake Error in src/ds++/CMakeLists.txt:
  Xcode does not support per-config per-source COMPILE_DEFINITIONS:

    BUILD_TYPE="$<CONFIG>"

  specified for source:

    /Users/clevelam/development/draco/src/ds++/Release.cc

CMake Error in src/ds++/CMakeLists.txt:
  Xcode does not support per-config per-source COMPILE_DEFINITIONS:

    BUILD_TYPE="$<CONFIG>"

  specified for source:

    /Users/clevelam/development/draco/src/ds++/Release.cc

CMake Error in src/ds++/CMakeLists.txt:
  Xcode does not support per-config per-source COMPILE_DEFINITIONS:

    BUILD_TYPE="$<CONFIG>"

  specified for source:

    /Users/clevelam/development/draco/src/ds++/Release.cc

CMake Warning:
  Manually-specified variables were not used by the project:

    USE_OPENMP
```

--- a/src/ds++/CMakeLists.txt
+++ b/src/ds++/CMakeLists.txt
@@ -79,8 +79,14 @@ list( APPEND headers
    ${PROJECT_BINARY_DIR}/ds++/dll_declspec.h )
 list( REMOVE_ITEM headers ${template_implementations} )
 
-set_source_files_properties( Release.cc PROPERTIES 
-  COMPILE_DEFINITIONS BUILD_TYPE="$<CONFIG>" )
+if( ${CMAKE_GENERATOR} MATCHES "Xcode" )
+  # Xcode does not support per file, per configuration compile definitions, so
+  # we add this compile definition for all files
+  add_compile_definitions( BUILD_TYPE="$<CONFIG>" )
+else()
+  set_source_files_properties( Release.cc PROPERTIES
+    COMPILE_DEFINITIONS BUILD_TYPE="$<CONFIG>" )
+endif()
 
 # ---------------------------------------------------------------------------- #
 # Build package library

